qr_generator/lib/screens/qr_details/bloc/qr_details_bloc.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:bloc/bloc.dart';
+import 'package:connectivity_plus/connectivity_plus.dart';
+import 'package:equatable/equatable.dart';
+import 'package:qr_generator/data/storage.dart';
+import 'package:qr_generator/model/seed_model.dart';
+import 'package:qr_generator/repositories/seed_repository.dart';
+
+part 'qr_details_event.dart';
+part 'qr_details_state.dart';
+
+class QrDetailsBloc extends Bloc<QrDetailsEvent, QrDetailsState> {
+  final SeedRepository _seedRepository;
+  final SuperformulaStorage _storage;
+
+  QrDetailsBloc(SeedRepository seedRepository, SuperformulaStorage storage)
+      : assert(seedRepository != null),
+        _seedRepository = seedRepository,
+        _storage = storage,
+        super(QrDetailsLoading());
+
+  @override
+  Stream<QrDetailsState> mapEventToState(QrDetailsEvent event) async* {
+    //Inital event
+    if (event is FetchSeed) {
+      yield QrDetailsLoading();
+      try {
+        yield await manageConnectivityData();
+      } catch (e) {
+        yield QrDetailsError(e.toString());
+      }
+    }
+    //Events after the app its load in the first time
+    var currentState = state;
+    if (currentState is QrDetailsLoaded) {
```

Comment:
**suggestion**: while this work, the body of `mapEventToState` (which will shortly be deprecated with bloc 8.0.0) isn't meant to be used like this. You should **react** to events and store your "result" in the state.

To remember that some data had already been loaded, use a variable in the state class. In your case, you should define some sort of bool variable in `QrDetailsState` and pass it down the hierarchy

Reply:
Thank you for the feedback, I'll read the flutter_bloc  migration documents.